sklearn/partial_dependence.py

```diff
@@ -0,0 +1,648 @@
+            pdp = pdp.ravel()
+        if pdp.ndim != 1 and pdp.shape[1] != 1:
+            # Multi-output
+            if not 0 <= output < pdp.shape[1]:
+                raise ValueError('Valid output must be specified for '
+                                 'multi-output models.')
+            pdp = pdp[:, output]
+        pdp = pdp[np.newaxis]
+    elif est._estimator_type == 'classifier':
+        try:
+            pdp = est.predict_proba(X_eval)
+        except NotFittedError:
+            raise ValueError('Call %s.fit before partial_dependence' %
+                             est.__class__.__name__)
+        if isinstance(pdp, list):
+            # Multi-output
+            if not 0 <= output < len(pdp):
+                raise ValueError('Valid output must be specified for '
+                                 'multi-output models.')
+            pdp = pdp[output]
+        pdp = np.log(np.clip(pdp, 1e-16, 1))
+        pdp = np.subtract(pdp, np.mean(pdp, 1)[:, np.newaxis])
+        pdp = pdp.transpose()
+    else:
+        raise ValueError('est must be a fitted regressor or classifier model.')
+    if pdp.shape[0] == 2:
+        # Binary classification
+        pdp = pdp[1, :][np.newaxis]
+    return pdp
+
+
+def partial_dependence(est, target_variables, grid=None, X=None, output=None,
+                       percentiles=(0.05, 0.95), grid_resolution=100,
+                       method=None):
+    """Partial dependence of ``target_variables``.
+
+    Partial dependence plots show the dependence between the joint values
+    of the ``target_variables`` and the function represented
+    by the ``est``.
+
+    Read more in the :ref:`User Guide <partial_dependence>`.
+
+    Parameters
+    ----------
+    est : BaseEstimator
+        A fitted classification or regression model.
+    target_variables : array-like, dtype=int
+        The target features for which the partial dependency should be
+        computed (size should be smaller than 3 for visual renderings).
+    grid : array-like, shape=(n_points, len(target_variables))
+        The grid of ``target_variables`` values for which the
+        partial dependency should be evaluated (either ``grid`` or ``X``
+        must be specified).
+    X : array-like, shape=(n_samples, n_features)
+        The data on which ``est`` was trained. It is used to generate
+        a ``grid`` for the ``target_variables``. The ``grid`` comprises
+        ``grid_resolution`` equally spaced points between the two
+        ``percentiles``.
+    output : int, optional (default=None)
+        The output index to use for multi-output estimators.
+    percentiles : (low, high), default=(0.05, 0.95)
+        The lower and upper percentile used to create the extreme values
+        for the ``grid``. Only if ``X`` is not None.
+    grid_resolution : int, default=100
+        The number of equally spaced points on the ``grid``.
+    method : {'recursion', 'exact', 'estimated', None}, optional (default=None)
```

Comment:
call it 'auto' instead of None?

I generally am against "default=None" because it doesn't tell you what it means. Certainly it is redundant with "optional".